docs/advanced/ObjectSpreadOperator.md

```diff
@@ -0,0 +1,37 @@
+# Object Spread Operator
+
+Since one of the core tenants of Redux is to never mutate state, you'll often find yourself using `Object.assign` to create
+copies of objects with new or updated values. For example, in the `todoApp` below `Object.assign` is used to return a new
+`state` object with an updated `visibilityFilter` property:
+
+```js
+function todoApp(state = initialState, action) {
+  switch (action.type) {
+    case SET_VISIBILITY_FILTER:
+      return Object.assign({}, state, {
+        visibilityFilter: action.filter
+      })
+    default:
+      return state
+  }
+}
+```
+
+While effective, using `Object.assign` can quickly make simple reducers difficult to read given it's rather verbose syntax.
+
+An alternative approach is to use the [object spread syntax](https://github.com/sebmarkbage/ecmascript-rest-spread) proposed for ES7
+which lets you use the spread (`...`) operator to copy enumerable properties from one object to another in a more succinct way. We
+can simplify the `todoApp` example above by using the object spread syntax:
+
+```js
+function todoApp(state = initialState, action) {
+  switch (action.type) {
+    case SET_VISIBILITY_FILTER:
+      return {...state, visibilityFilter: action.filter}
+    default:
+      return state
+  }
+}
+```
+
```

Comment:
We would really benefit from a few more examples showing strong sides of the spread operator.
For example:

```js
  return getAddedIds(state.cart).map(id => Object.assign(
    {},
    getProduct(state.products, id),
    {
      quantity: getQuantity(state.cart, id)
    }
  ))
```

vs

```js
  return getAddedIds(state.cart).map(id => ({
    ...getProduct(state.products, id),
    quantity: getQuantity(state.cart, id)
  }))
```